### PR TITLE
📝 Update release documentation

### PR DESCRIPTION
**Pull request by r-richmond**
Updates the release documentation based on the slack conversation [here](https://apache-airflow.slack.com/archives/C03G9H97MM2/p1685467707791689?thread_ts=1685466670.125099&cid=C03G9H97MM2) to note the following items
* PRs that are targeted for a patch version release are generally marked with the corresponding milestone reference (but deviations happen due to the manual process)
* Maintainers reserve the right to punt a PR to a subsequent release at any time
* New features are not allowed into patch releases and must wait for the next minor release (which is currently targeted at every 2-3 months).

### CODE REVIEW

File: docs/apache-airflow/release-process.rst [hunk 1]
```diff
@@ -30,6 +30,9 @@ Since Airflow 2.0.0 and provider packages 1.0.0 we aim to follow SemVer, meaning
 
 In git, each minor version will have its own branch, called ``vX-Y-stable`` where bugfix/security releases will be issued from.
 Commits and PRs should not normally go direct to these branches, but instead should target the main branch and then be cherry-picked by the release managers to these release branches.
+As a general rule of thumb changes that will be included in a bugfix/security release will be associated with the corresponding github milestone in the PR but this is currently a manual process and deviations may occur.
+In all cases the maintainers reserve the right to punt a PR to a later release if they deem it prudent.
```
Comment (ephraimbuddy) [inline docs/apache-airflow/release-process.rst]:
```suggestion
In all cases the release manager reserve the right to punt a PR to a later release if they deem it prudent.
```

**Review by ephraimbuddy — APPROVED** [review]

File: docs/apache-airflow/release-process.rst [hunk 2]
```diff
@@ -30,6 +30,9 @@ Since Airflow 2.0.0 and provider packages 1.0.0 we aim to follow SemVer, meaning
 
 In git, each minor version will have its own branch, called ``vX-Y-stable`` where bugfix/security releases will be issued from.
 Commits and PRs should not normally go direct to these branches, but instead should target the main branch and then be cherry-picked by the release managers to these release branches.
+As a general rule of thumb changes that will be included in a bugfix/security release will be associated with the corresponding github milestone in the PR but this is currently a manual process and deviations may occur.
```
Comment (pankajkoti) [inline docs/apache-airflow/release-process.rst]:
```suggestion
As a general rule of thumb, changes that will be included in a bugfix/security release will be associated with the corresponding github milestone in the PR but this is currently a manual process and deviations may occur.
```

File: docs/apache-airflow/release-process.rst (hunk 1 above)
Comment (pankajkoti) [inline docs/apache-airflow/release-process.rst]:
```suggestion
In all cases, the release managers reserve the right to postpone a PR to a later release if they deem it prudent.
```

Readers like me might not have a strong vocabulary, so suggesting punt -> postpone if you'd like to consider :)

**Review by pankajkoti — APPROVED** [review]

File: docs/apache-airflow/release-process.rst (hunk 1 above)
Comment (r-richmond) [reply docs/apache-airflow/release-process.rst]: resolved in https://github.com/apache/airflow/pull/31631/commits/1a9c7a3ec244e0b4a8c6cc1988d7398e6aa143f5

**Review by jedcunningham — APPROVED** [review]

**Pull request merged**